web/client/components/TOC/fragments/Title.jsx

```diff
@@ -38,14 +38,19 @@ class Title extends React.Component {
     };
 
     getFilteredTitle = (title) => {
-        const splitTitle = title.toLowerCase().split(this.props.filterText.toLowerCase());
+        const regularExpression = new RegExp(this.props.filterText.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&'), 'gi');
+        const matches = title.match(regularExpression);
 
-        return this.props.filterText ? splitTitle.reduce((a, b, idx) => {
-            if (idx === splitTitle.length - 1) {
-                return [...a, b];
+        if (!this.props.filterText || !matches) {
+            return title;
+        }
+
+        return title.split(regularExpression).map((split, key) => {
+            if (key < matches.length) {
+                return [...split, <strong>{matches[key]}</strong>];
```

Comment:
i would call the index idx

```suggestion
        return title.split(regularExpression).map((split, idx) => {
            if (idx < matches.length) {
                return [...split, <strong key={idx}>{matches[idx]}</strong>];
```